### PR TITLE
Comments for queuing of transformers/requests

### PR DESCRIPTION
**Pull request by sven-urbanski-freiheit-com**

### CODE REVIEW

File: services/cd-service/pkg/repository/repository.go [hunk 1]
```diff
@@ -471,12 +473,23 @@ func (r *repository) Apply(ctx context.Context, transformers ...Transformer) err
...
 		r.maybeGc(ctx)
 	}()
-	eCh := r.applyDeferred(ctx, transformers...)
-	select {
-	case err := <-eCh:
-		return err
-	case <-ctx.Done():
-		return ctx.Err()
+	const deferIt = true
+	if deferIt {
```
Comment (hannesg) [inline services/cd-service/pkg/repository/repository.go:477]: Is this intended to be here?

File: README_DEVELOPERS.md
Comment (tameremad) [inline README_DEVELOPERS.md:1]: I like the clarity here :+1:

File: services/cd-service/pkg/repository/repository.go (hunk 1 above)
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/repository.go:477]: no!

Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/repository.go]: I removed it.
(that was an attempt to compare the benchmark between deferred/non-deferred operations. It was not finished)

File: services/cd-service/pkg/repository/queue.go [hunk 2]
```diff
@@ -18,9 +18,10 @@ package repository
 
 /**
 This queue contains transformers. Do not confuse with the "queuedVersion" field in protobuf (api.proto).
-The queue here is used because applying a change to git (pushing) takes some time,
-but this violates general rest patterns (every endpoint should return quickly).
-So we queue transformers, and don't wait for them to finish.
+The queue here is used because applying a change to git (pushing) takes some time.
+Still, every request waits for the transformer AND push to finish (that's what the `result` channel is for).
```
[the remark this answers is not in the document]
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/queue.go]: I put in some more details in the comments. Not sure if anything's missing still.

Latency: Duration from request start to request end.
Throughput: How many requests can we handle in a given time frame.

**Review by hannesg — APPROVED** [review]

**Pull request merged**